components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/TransactionHandlerComponent.java

```diff
@@ -0,0 +1,172 @@
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.wso2.micro.integrator.initializer.handler.transaction;
+
+import org.apache.axis2.transport.base.BaseConstants;
+import org.apache.axis2.util.JavaUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.config.mapper.ConfigParser;
+import org.wso2.micro.core.util.StringUtils;
+import org.wso2.micro.integrator.initializer.handler.transaction.security.CryptoUtil;
+import org.wso2.micro.integrator.initializer.handler.transaction.store.TransactionStore;
+import org.wso2.micro.integrator.ndatasource.common.DataSourceException;
+import org.wso2.micro.integrator.ndatasource.core.CarbonDataSource;
+import org.wso2.micro.integrator.ndatasource.core.DataSourceService;
+
+import java.util.UUID;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.crypto.Cipher;
+import javax.sql.DataSource;
+
+/**
+ * This class represents the internal transaction count handler component.
+ */
+public class TransactionHandlerComponent {
+
+    private static final Log LOG = LogFactory.getLog(TransactionHandlerComponent.class);
+    private static final int DEFAULT_SCHEDULED_PERIOD_INTERVAL = 60;
+    private static final String SCHEDULED_PERIOD = "transactionCountScheduledPeriod";
+
+    private static final String TRANSACTION_DATA_SOURCE = "transactionDataSource";
+    private static final String TRANSACTION_CONFIG_SECTION = "transaction";
+    private static final String TRANSACTION_CONFIG_NAME = "data_source";
+
+    public void start(DataSourceService dataSourceService) {
+        String nodeId = generateRandomId();
+        try {
+            // initialize cipher for encryption needs.
+            Cipher cipher = CryptoUtil.initializeCipher();
+
+            DataSource dataSource = getTransactionDataSource(dataSourceService);
+
+            // initialize Transaction store.
+            TransactionStore transactionStore = new TransactionStore(dataSource, nodeId, cipher);
+
+            // start and schedule transaction writer task.
+            scheduleTransactionWriterTask(transactionStore);
+
+        } catch (DataSourceException | TransactionException e) {
+            LOG.error("Error in initializing Transaction Service component: " + e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Get the time rate for running the transaction writer scheduled task.
+     *
+     * @return - scheduled period.
+     */
+    private int getTransactionCountScheduledPeriod() {
+        String scheduledPeriodStr = System.getProperty(SCHEDULED_PERIOD);
+        if (StringUtils.isEmpty(scheduledPeriodStr)) {
+            scheduledPeriodStr = System.getenv(SCHEDULED_PERIOD);
+        }
+        if (StringUtils.isEmpty(scheduledPeriodStr)) {
```

Comment:
Shall we move this inside above if block.

Reply:
Done.